src/components/Select/Select.tsx

```diff
@@ -0,0 +1,130 @@
+import { SelectProps } from './Select.types';
+import React, { useState, useMemo } from 'react';
+import {
+  FormControl,
+  Select as MuiSelect,
+  MenuItem,
+  InputLabel,
+  ListSubheader,
+  TextField,
+  InputAdornment,
+  Divider,
+  Box,
+  CircularProgress,
+} from '@mui/material';
+import SearchIcon from '@mui/icons-material/Search';
+
+/**
+ * This component kets the user select an item from a dropdown.
+ *
+ * @return {JSX.Element}
+ */
+const Select: React.FC<SelectProps> = ({
+  options,
+  option,
+  fixedOptions,
+  setOption,
+  loading,
+}) => {
+  const containsText = (text: string, searchText: string) =>
+    text.toLowerCase().indexOf(searchText.toLowerCase()) > -1;
+
+  const [searchText, setSearchText] = useState('');
+
+  const displayedOptions = useMemo(
+    () =>
+      options
+        .filter((option) => !fixedOptions?.includes(option))
+        .filter((option) => containsText(option, searchText)),
+    [options, fixedOptions, searchText]
+  );
+
+  const displayedFixedOptions = useMemo(
+    () =>
+      fixedOptions?.filter((option) => containsText(option, searchText)) ?? [],
+    [searchText, fixedOptions]
+  );
+
+  const displaySelect =
+    (fixedOptions != undefined && fixedOptions.length > 0) ||
+    options.length > 0 ||
+    loading;
+
+  return (
+    <>
+      {displaySelect ? (
+        <FormControl fullWidth>
+          <InputLabel id="search-select-label">Language</InputLabel>
```

Comment:
So this is a bit of duplicate from the previous comment. If we are using Language here, then it becomes quite specific component, like maybe LanguageSelector.

Reply:
I've added a prop for the label to keep it generic.